eyecite/resolve.py

```diff
@@ -148,10 +200,10 @@ def resolve_citations(
     resolved.
     """
     # Dict of all citation resolutions
-    resolutions: Dict[ResourceType, List[CitationBase]] = defaultdict(list)
+    resolutions: Resolutions = defaultdict(list)
 
     # Dict mapping full citations to their resolved resources
-    resolved_full_cites: Dict[FullCitation, ResourceType] = {}
+    resolved_full_cites: ResolvedFullCites = []
```

Comment:
What's the idea behind switching from a dict to a list of tuples here?

Reply:
The purpose of the dict was to remove duplicates in this block:

```
        if isinstance(citation, FullCitation):
            resolution = resolve_full_citation(citation)
            resolved_full_cites[citation] = resolution
```

... but that wasn't actually removing any duplicates, because identical cites from different parts of the text aren't considered duplicates for hash equality. I got partway through implementing something that would remove duplicates based on semantic equality (and I kept the useful parts of that effort in `models.py`), but then I realized that I was complicating this part of the code for no reason, because duplicates are rare and already need to be handled further down the line anyway, which is why we didn't notice that they weren't being handled here in the first place.

So at that point `resolved_full_cites` was just a list in dict form, and I made it an actual list to make its function more clear.